### PR TITLE
Logging for JsonRpcMiddleware is now separated so it can be easily overriden in derived classes.

### PR DESCRIPTION
**Pull request by RafalSzefler**

### CODE REVIEW

**Comment by matee911** [comment]
@RafaelSzefler +1 but wait for @gacu82

File: src/JsonRpc.Host/JsonRpcMiddleware.cs [hunk 1]
```diff
@@ -65,7 +83,6 @@ internal class JsonRpcMiddleware
                 responseString = await processor.ProcessAsync(requestString, service);
                 diagnostics.ProcessTime = watch.ElapsedMilliseconds - diagnostics.ReadTime;
                 response.Headers.Add("Content-Type", "application/json");
-                response.Headers.Add("X-ProcessingTime", diagnostics.ProcessTime.ToString());
```
Comment (ekamil) [inline src/JsonRpc.Host/JsonRpcMiddleware.cs:68]: This doesn't seem related to the commit message

Comment (RafalSzefler) [reply src/JsonRpc.Host/JsonRpcMiddleware.cs:68]: Yeah. Removed because this is sort of an information leak. Does not seem to be of any importance to frontend and yet may be used for some malicious purpose (e.g. timing attack).
